docs/content/preview/releases/techadvisories/ta-23476.md

```diff
@@ -0,0 +1,169 @@
+
+Additionally, if millisecond precision is not sufficient for your use case, there are two options:
+
+1. Switch from timestamp to timeuuid data type:
+
+    1. Timeuuid uses 100-nanosecond precision and includes a node-identifier component, ensuring values generated on different nodes can never collide.
+
+1. If millisecond-precision collisions are expected but rare:
+
+    1. Use IF NOT EXISTS on the data insert path and handle the existing row case in the application by retrying the insert. However, if collisions are common, this approach may result in too many retries.
+
+## Details
+
+The YCQL wire protocol uses millisecond precision for timestamps, while the YCQL storage system uses microsecond precision internally. This means that all timestamps that can be inserted are expected to have `000` for the microseconds component.
+
+However, the `currenttimestamp()` function currently generates values with microsecond precision, so the microseconds component is likely to be non-zero 99.9% of the time.
+
+When these non-zero microsecond timestamp values are stored, the microsecond information will be lost when the precision is adjusted to milliseconds for the wire protocol (for example, when responding to a user query). As a result, you will always see timestamp values with only millisecond precision.
+
+This can create issues in certain scenarios as follows:
+
+- Equality conditions: If you query for a timestamp value that was just read, the condition will not match, as the microsecond component will differ (0 for the query, non-zero for the stored value).
+
+- Index scans: If the primary key of the main table includes an affected timestamp column, the data in both the table and the index will be in microsecond precision, so they will match. However, when performing an index scan, the main-table read request uses the same protocol as the YCQL API, which adjusts the precision to milliseconds. This may cause the main-table read request to not match, and the index scan result to miss the affected rows with non-zero microsecond values.
+
+## Examples
+
+### Primary-key scan with equality condition
+
+1. Set up the keyspace and table as follows:
+
+    ```cql
+    CREATE KEYSPACE temp;
+    USE temp;
+    CREATE TABLE test(id int, ts timestamp, v1 int, v2 int, primary key( id, ts)) WITH transactions = {'enabled': 'true'};
+    ```
+
+1. Insert a row using `currenttimestamp()` as follows:
+
+    ```cql
+    INSERT INTO test(id, ts, v1, v2) VALUES (1, currenttimestamp(), 5, 10);
+    SELECT * FROM test;
+    ```
+
+    ```output
+     id | ts                              | v1 | v2
+    ----+---------------------------------+----+----
+      1 | 2024-08-21 17:44:47.647000+0000 |  5 | 10
+    ```
+
+    As expected the row is present. However, most likely the internal row has a non-zero microsecond component, just that due to the YCQL API millisecond standard, it will always return `000` for the microsecond part.
```

Comment:
```suggestion
    As expected, the row is present. However, most likely the internal row has a non-zero microsecond component, just that due to the YCQL API millisecond standard, it will always return `000` for the microsecond part.
```